cetmix_tower_server/models/cx_tower_file_template.py

```
@@ -136,45 +136,71 @@ def create_file(self, server, server_dir="", raise_if_exists=False):
                 ("server_dir", "=", server_dir or self.server_dir),
                 ("source", "=", self.source),
             ],
-            limit=1,
+            order="id DESC",
         )
-        if existing_file:
-            if raise_if_exists:
-                raise ValidationError(_("File already exists on server."))
+        # Filter existing files by name or rendered name
+        existing_file_ids = existing_file_ids.filtered(
+            lambda f: f.name in self.file_name or f.rendered_name in self.file_name
```

Flake8 (7.2.0)

[error] line too long (83 > 79 characters)

(E501)